3rdparty/btfhub.sh

```diff
@@ -0,0 +1,100 @@
+#!/bin/bash
+
+# This script downloads & updates 2 repos inside tracee dir structure:
+#
+#     1) ./3rdparty/btfhub-archive
+#     2) ./3rdparty/btfhub
+#
+# It uses the 2 repositories to generate tailored BTF files, according to
+# latest CO-RE object file, so those files can be embedded in tracee Go binary.
+
+BASEDIR=$(dirname "${0}") ; cd ${BASEDIR}/../ ; BASEDIR=$(pwd) ; cd ${BASEDIR}
+
+# variables
+
+TRACEE_BPF_CORE="${BASEDIR}/dist/tracee.bpf.core.o"
+BTFHUB_REPO="https://github.com/aquasecurity/btfhub.git"
+BTFHUB_ARCH_REPO="https://github.com/aquasecurity/btfhub-archive.git"
+BTFHUB_DIR="${BASEDIR}/3rdparty/btfhub"
+BTFHUB_ARCH_DIR="${BASEDIR}/3rdparty/btfhub-archive"
+
+ARCH=$(uname -m)
+
+case ${ARCH} in
+    "x86_64")
+        ARCH="x86_64"
+        NOT_ARCH=("arm64")
+        ;;
+    "aarch64")
+        ARCH="arm64"
+        NOT_ARCH=("x86_64")
+        ;;
+    *)
+        die "unsupported architecture" ;;
+esac
+
+die() {
+    echo ${@}
+    exit 1
+}
+
+branch_clean() {
+    cd ${1} || die "could not change dirs"
+
+    # small sanity check
+    [ ! -f ./README.md ] && die "$(basename $(pwd)) not a repo dir"
+
+    git fetch -a || die "could not fetch ${1}"  # make sure its updated
+    git clean -fdX                              # clean leftovers
+    git reset --hard                            # reset letfovers
+    git checkout origin/main -b main-$$
+    git branch -D main
+    git branch -m main-$$ main                  # origin/main == main
+
+    cd ${BASEDIR}
+}
+
+[ ! -f ${TRACEE_BPF_CORE} ] && die "tracee CO-RE obj not found"
+
+[ ! -d ${BTFHUB_DIR} ] && git clone "${BTFHUB_REPO}" ${BTFHUB_DIR}
+[ ! -d ${BTFHUB_ARCH_DIR} ] && git clone "${BTFHUB_ARCH_REPO}" ${BTFHUB_ARCH_DIR}
+
+branch_clean ${BTFHUB_DIR}
+branch_clean ${BTFHUB_ARCH_DIR}
+
+cd ${BTFHUB_DIR}
+
+# only supported distros and kernels
```

Comment:
Im excluding those here because they already support embedded BTF and I want to include only BTFs for "really needed" environments.